libs/components/indicators/src/lib/modules/wait/wait.service.ts

```diff
@@ -26,17 +33,21 @@ export class SkyWaitService {
   #appRef: ApplicationRef;
   #waitAdapter: SkyWaitPageAdapterService;
   #windowSvc: SkyAppWindowRef;
+  #dynamicComponentService: SkyDynamicComponentService | undefined;
 
+  // TODO: Remove `Optional` in a future breaking change. When this is done - remove the `resolver` variable and any cases that account for it.
   constructor(
     resolver: ComponentFactoryResolver,
     appRef: ApplicationRef,
     waitAdapter: SkyWaitPageAdapterService,
-    windowSvc: SkyAppWindowRef
+    windowSvc: SkyAppWindowRef,
+    @Optional() dynamicComponentService: SkyDynamicComponentService
```

Comment:
This is optional for now since this is a service - I can create a story to remove it in 8 once we merge this. If we don't make it optional it would break consumers in a manner similar to what the overload did.

Reply:
I did try to make comments anywhere code will need to change once we make this non-optional.